packages/rich-list-core/src/RichListCore.ts

```diff
@@ -0,0 +1,219 @@
+import { ApiClient, BigNumber, blockchain as defid } from '@defichain/jellyfish-api-core'
+import { QueueClient, Queue } from './lib/Queue'
+import { SingleIndexDb, Schema } from './lib/SingleIndexDb'
+import { AddressParser } from './controller/AddressParser'
+import { NetworkName } from '@defichain/jellyfish-network'
+import { RichListItem } from '@defichain/rich-list-api-client'
+import { AccountAmount } from '@defichain/jellyfish-api-core/src/category/account'
+import { ActiveAddressAccountAmount } from './controller/AddressParser/ActiveAddressAccountAmount'
+
+const DEFAULT_RICH_LIST_LENGTH = 1000
+
+export class RichListCore {
+  isCatchingUp = false
+  richListLength = DEFAULT_RICH_LIST_LENGTH
+  readonly addressParser: AddressParser
+
+  constructor (
+    private readonly network: NetworkName,
+    private readonly apiClient: ApiClient,
+    private readonly existingRichList: SingleIndexDb<RichListItem>,
+    private readonly crawledBlockHashes: SingleIndexDb<CrawledBlock>,
+    private readonly droppedFromRichList: SingleIndexDb<string>,
+    private readonly queueClient: QueueClient<string>
+  ) {
+    this.addressParser = new AddressParser(apiClient, network)
+  }
+
+  setRichListLength (length: number): void {
+    this.richListLength = length
+  }
+
+  resume (): void {
+    if (this.isCatchingUp) {
+      return
+    }
+    this.isCatchingUp = true
+    void this._catchUp()
+  }
+
+  private async _catchUp (): Promise<void> {
+    const lastBlock = await this._getCrawledTip()
+    const nextBlockHeight = lastBlock?.data.height ?? 0
```

Comment:
hmm.. looking for nextblock based on lastblock.. isn't it `nextblockheight = lastblockheight + 1`??

Reply:
Yes you're right, it is a mistake. The previous implementation using a `LinkedList` like interface where `size()` is equivalent to previous height + 1. I should add +1 here.